bin/ch/PlatformAgnostic/Platform/Windows/Module.cpp

```diff
@@ -0,0 +1,52 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+#include "PlatformAgnostic/ChPlatformAgnostic.h"
+#include "Common.h"
+#include <strsafe.h>
+
+namespace ChPlatformAgnostic
+{
+    bool Module::GetBinaryLocation(char* const path, const charcount_t size, charcount_t* const resultStrLength)
+    {
+        size_t bufferSize = sizeof(WCHAR) * size;
+
+        if (path == nullptr || bufferSize < size) 
```

Comment:
Maybe I misunderstand the code, but is `bufferSize < size` necessary? Because here `bufferSize` is always `sizeof(WCHAR) * size`, which is larger than `size`. Also it seems `bufferSize` isn't used after this line.

Reply:
It's checking for overflow

Reply:
Thanks! I didn't realize it's an integer overflow checking. Sorry!